new data source for MongoDB instances

malferov opened the pull request:
new data source for MongoDB instances and its documentation

xiaozhu36 commented on alicloud/data_source_alicloud_mongo_instances.go:
```diff
@@ -0,0 +1,263 @@
+			break
+		}
+
+		args.PageNumber = args.PageNumber + requests.NewInteger(1)
```
HI @malferov line 206 is a bug and you can refer to https://github.com/terraform-providers/terraform-provider-alicloud/pull/250/commits/7019687b9931070c00a0281d4f39371990729b26#diff-90d96cd66a3012b469997eb1cbceb593R132

malferov replied:
fixed

xiaozhu36 commented on alicloud/services/dds/client.go:
```diff
@@ -0,0 +1,108 @@
+package dds
```
I will push mongoDB developer to publish official go sdk. Please be patient.

xiaozhu36 replied:
Please use official go sdk: https://github.com/aliyun/alibaba-cloud-sdk-go/tree/master/services/dds

xiaozhu36 commented on alicloud/data_source_alicloud_mongo_instances.go:
```diff
@@ -0,0 +1,269 @@
+					"MongoDB",
+				}),
+			},
+			"db_type": {
```
I think line 30 should be instance_type

xiaozhu36 commented on alicloud/data_source_alicloud_mongo_instances.go:
```diff
@@ -0,0 +1,269 @@
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"db_type": {
```
what is db_type

xiaozhu36 commented on alicloud/data_source_alicloud_mongo_instances.go:
```diff
@@ -0,0 +1,269 @@
+								},
+							},
+						},
+						"shard": {
```
I think line 129 should be shards

malferov commented:
@xiaozhu36 updated and switched to official go sdk. please review

xiaozhu36 commented on alicloud/data_source_alicloud_mongo_instances.go:
```diff
@@ -0,0 +1,269 @@
+				Optional:     true,
+				ValidateFunc: validateNameRegex,
+			},
+			"engine": {
```
If engine only has one validate value, the field can be removed and new flied 'availability_zone' and 'instance_class' can be added.

xiaozhu36 commented:
@malferov Thanks for your contribution. Can you rebase all of commits and remove useless annotation before the next submitting?

malferov commented:
@xiaozhu36 indeed `engine` argument is useless due to one possible value so far. so i have removed that and added two more according your advice above. the branch has been rebased. thanks
